core/rewards/staking_reward_calculator.go

```diff
@@ -180,3 +178,24 @@ func calculateRewardsByStake(epochSeq, asset, accountID string, rewardBalance *n
 		epochSeq:      epochSeq,
 	}
 }
+
+func adjustWeights(delegatorWeights map[string]num.Decimal, weightSums num.Decimal, sortedParties []string, decimalOne num.Decimal, rng *rand.Rand) {
+	// NB: due to rounding errors this sum can be greater than 1
+	// to avoid overflow, we choose at random a party adjust it by the error
+	// we keep looking until we find a candidate with sufficient weight to adjust by the precision error
+	for weightSums.GreaterThan(decimalOne) {
+		precisionError := weightSums.Sub(decimalOne)
+		for precisionError.GreaterThan(num.DecimalZero()) {
+			unluckyParty := sortedParties[rng.Intn(len(delegatorWeights))]
```

Comment:
Would it be simpler / safer to take the precision error from the party with the biggest weight rather than choosing a random one?

Reply:
it will be equally safe, and slightly less fair, but I don't mind

Reply:
Sorry, I was missing the outer `for` loop in my thinking. We go and try to remove the `precisionError` repeatedly till it's gone. So the case I am describing above isn't a problem.

Reply:
Let's go with the random integer selection.

Reply:
though equally safe and slightly less fair I would actually prefer to go with adjusting the largest. For the largest it would probably be a meaningless adjustment so I don't think it matters.